RenderStateNotation/src/RenderStateNotationParserImpl.cpp

```diff
@@ -359,8 +401,18 @@ Bool RenderStateNotationParserImpl::ParseString(const Char* StrData, Uint32 Leng
                     Deserialize(Json, ResourceDesc, Allocator);
                     VERIFY_EXPR(ResourceDesc.Name != nullptr);
 
-                    m_RenderPassNames.emplace(HashMapStringKey{ResourceDesc.Name, false}, StaticCast<Uint32>(m_RenderPasses.size()));
-                    m_RenderPasses.push_back(ResourceDesc);
+                    auto const Iter = m_RenderPassNames.find(ResourceDesc.Name);
+                    if (Iter != m_RenderPassNames.end())
+                    {
+                        if (!(m_RenderPasses[Iter->second] == ResourceDesc))
+                            LOG_ERROR_AND_THROW("Redefinition of render pass '", ResourceDesc.Name, "'.");
+                    }
+                    else
+                    {
+                        m_RenderPassNames.emplace(HashMapStringKey{ResourceDesc.Name, false}, StaticCast<Uint32>(m_RenderPasses.size()));
```

Comment:
You can check `emplace(..).second` to see if the element has been actually inserted to avoid searching in the has map two times: first time to check it is present and second time to actually insert it.
Most of the time there will be no duplicate elements.